Modules/Loadable/Volumes/SubjectHierarchyPlugins/qSlicerSubjectHierarchyVolumesPlugin.cxx

```diff
@@ -1298,3 +1334,36 @@ void qSlicerSubjectHierarchyVolumesPlugin::setVolumePreset(const QString& preset
     volumesModuleLogic->ApplyVolumeDisplayPreset(d->SelectedVolumeNode->GetVolumeDisplayNode(), presetId.toStdString());
     }
 }
+
+//---------------------------------------------------------------------------
+void qSlicerSubjectHierarchyVolumesPlugin::toggleVisibilityForCurrentItem(bool on)
+{
+  Q_D(qSlicerSubjectHierarchyVolumesPlugin);
+  if (d->SelectedVolumeNode)
+    {
+    // Show color legend display node
+    vtkMRMLDisplayNode* displayNode = d->SelectedVolumeNode->GetDisplayNode();
```

Comment:
You need to call `GetVolumeDisplayNode()` to make sure you don't get a volume rendering or color legend display node.

```suggestion
    vtkMRMLDisplayNode* displayNode = d->SelectedVolumeNode->GetVolumeDisplayNode();
```